core/modules/replica/ConfigurationMySQL.cc

```diff
@@ -238,133 +255,149 @@ void ConfigurationMySQL::loadConfiguration() {
-
-        std::string table;
-        ::readMandatoryParameter(row, "table", table);
+    database::mysql::Connection::Ptr conn;
 
-        bool isPartitioned;
-        ::readMandatoryParameter(row, "is_partitioned", isPartitioned);
+    try {
+        conn = database::mysql::Connection::open(_connectionParams);
+        conn->execute(
+            [&](decltype(conn) conn) {
+    
+                database::mysql::Row row;
+            
+                // Read the common parameters and defaults shared by all components
+                // of the replication system. The table also provides default values
+                // for some critical parameters of the worker-side services.
+            
+                conn->execute("SELECT * FROM " + conn->sqlId("config"));
+            
+                while (conn->next(row)) {
+            
+                    ::tryParameter(row, "common", "request_buf_size_bytes",     _requestBufferSizeBytes) or
+                    ::tryParameter(row, "common", "request_retry_interval_sec", _retryTimeoutSec) or
+            
+                    ::tryParameter(row, "controller", "num_threads",         _controllerThreads) or
+                    ::tryParameter(row, "controller", "http_server_port",    _controllerHttpPort) or
+                    ::tryParameter(row, "controller", "http_server_threads", _controllerHttpThreads) or
+                    ::tryParameter(row, "controller", "request_timeout_sec", _controllerRequestTimeoutSec) or
+                    ::tryParameter(row, "controller", "job_timeout_sec",     _jobTimeoutSec) or
+                    ::tryParameter(row, "controller", "job_heartbeat_sec",   _jobHeartbeatTimeoutSec) or
+            
+                    ::tryParameter(row, "database", "services_pool_size", _databaseServicesPoolSize) or
+            
+                    ::tryParameter(row, "xrootd", "auto_notify",         _xrootdAutoNotify) or
+                    ::tryParameter(row, "xrootd", "host",                _xrootdHost) or
+                    ::tryParameter(row, "xrootd", "port",                _xrootdPort) or
+                    ::tryParameter(row, "xrootd", "request_timeout_sec", _xrootdTimeoutSec) or
+            
+                    ::tryParameter(row, "worker", "technology",                 _workerTechnology) or
+                    ::tryParameter(row, "worker", "num_svc_processing_threads", _workerNumProcessingThreads) or
+                    ::tryParameter(row, "worker", "num_fs_processing_threads",  _fsNumProcessingThreads) or
+                    ::tryParameter(row, "worker", "fs_buf_size_bytes",          _workerFsBufferSizeBytes) or
+                    ::tryParameter(row, "worker", "svc_port",                   commonWorkerSvcPort)  or
+                    ::tryParameter(row, "worker", "fs_port",                    commonWorkerFsPort) or
+                    ::tryParameter(row, "worker", "data_dir",                   commonWorkerDataDir);
+                }
```

Comment:
It's a replacement for a C/C++ switch/case which doesn't support strings as keys in the 'case' statements :-) I have it here to reduce noise in the code an to make it readable. This patter has some performance penalty because (in the word case scenario) it has O(N*2). A good news is that N=10 (or so).
By the way, did you know that C/C++ 'switch' is actually implemented on  a similar way like a long:
`if (case1) {}`
`else if (case2) {}`
`else if (case3) {}`
`...`

Reply:
Map-based implementation would result in a significantly larger amount of code compared with the overloaded/templatized 'tryParameter'. Note that values of parameters parsed by this method have different types. If I were to use a map then I would probably have to introduce a hierarchy of classes (with subclasses handling specific types).
Efficiency is not an issue here because the number of parameters is rather small (less than 100). Besides, this algorithm runs just once at a startup of a service (master or workers).